File: javascript/matrix/matrix.js
function Matrix(stringizedMatrix) {
	this.input = stringizedMatrix;
	this.rows = this.parseRows();
	this.columns = this.parseColumns();
}

Matrix.prototype.parseRows = function() {
	var stringRows = this.input.split('\n');
	
	var rows = [];

	// REVIEW: I could have done this with a `for` and a `map()` call or two `for`s,
	// but for some reason JSLint complained about creating functions inside loops.
	stringRows.map(function(stringRow) {
		rows.push(stringRow.split(' ').map(function(element) {
			return parseInt(element);
		}));
	});
	
	return rows;
};

Matrix.prototype.parseColumns = function() {
	var columns = [];
	
	var colsNum = this.rows.length;
	var rowsNum = this.rows[0].length;
	for (var i = 0; i < rowsNum; ++i) {
		var column = [];
		
		for (var j = 0; j < colsNum; ++j) {
			column.push(this.rows[j][i]);
		}
		
		columns.push(column);
	}
	
	return columns;
};

module.exports = Matrix;

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, string_matrix):
        self.str_matrix = string_matrix
        self.rows = self.parse_rows()
        self.columns = self.parse_columns()
    
    def parse_rows(self):
        str_rows = self.str_matrix.split('\n')
        rows = []

        for row in str_rows:
            split_row = row.split(' ')
            new_row = []
            for item in split_row:
                new_row.append(int(item))
            
            rows.append(new_row)
        
        return rows
    
    def parse_columns(self):
        columns = []
    
        cols_num = len(self.rows)
        rows_num = len(self.rows[0])
        for i in range(rows_num):
            column = []
            
            for j in range(cols_num):
                column.append(self.rows[j][i])
            
            columns.append(column)
        
        return columns